lib/phabricator/phab_object.rb

```diff
@@ -2,18 +2,126 @@
 
 module Phabricator
   class PhabObject
-    attr_reader :phid
+    def self.inherited(other)
```

Comment:
Basically stuff like chalk-dsl, but I'm surprised there isn't stuff that is established off-the shelf open source work. Seems this type of generic and abstract code _should_ be a module you can just require and mixin.

In Perl, there's like 3…